catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -811,6 +814,53 @@ public void setDistanceTolerance(int distanceTolerance) {
     }
   }
 
+  public void setPollInterval(Integer interval) {
+    if (interval == null || interval.equals(pollInterval)) {
+      return;
+    }
+
+    this.pollInterval = interval;
+    LOGGER.debug(
+        "Setting Availability poll task for {} minute(s) on Source {}", pollInterval, getId());
+
+    if (scheduler != null) {
+      scheduler.shutdownNow();
+    }
+
+    scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("openSearchSourceThread"));
+
+    scheduler.scheduleWithFixedDelay(availabilityCheck, 0, pollInterval, TimeUnit.MINUTES);
+  }
+
+  public void destroy(int code) {
+    if (scheduler != null) {
+      scheduler.shutdownNow();
+    }
+  }
+
+  private class AvailabilityCheck implements Runnable {
+    volatile boolean isAvailable;
+
+    @Override
+    public void run() {
+      isAvailable = check();
+    }
+
+    private boolean check() {
+      LOGGER.debug("Checking availability for source {} ", getId());
```

Comment:
✏️ 	Trace seems more appropriate here
```suggestion
      LOGGER.trace("Checking availability for source {} ", getId());
```